### PR TITLE
feat: make probe always open blockdevices in readonly mode

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/go-blockdevice/issues/44

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

### CODE REVIEW

File: blockdevice/probe/probe.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func probe(devpath string) (devpaths []string) {
 	// Start by opening the block device.
 	// If a partition table was not found, it is still possible that a
 	// file system exists without a partition table.
-	bd, err := blockdevice.Open(devpath)
+	bd, err := blockdevice.Open(devpath, blockdevice.WithMode(blockdevice.ReadonlyMode))
```
Comment (smira) [inline blockdevice/probe/probe.go:152]: I hope we don't write to probed devices later? (I don't know if there's any good chance to track through it?)

**Review by smira — APPROVED** [review]

File: blockdevice/probe/probe.go (hunk 1 above)
Comment (Unix4ever) [reply blockdevice/probe/probe.go:152]: I've checked the code, it doesn't look like we do. I'm going to run integration tests in Talos now to see if anything got broken.

**Pull request merged**